File: 15-1.rb
#!/usr/bin/env ruby

class HASH
  attr_accessor :value

  def initialize(string)
    @string = string
    @value = 0
    string.codepoints.each do |c|
      @value = ((@value + c) * 17) % 256
    end
  end
end

strings = File.read('15.input').lines.map(&:strip)[0].split ','
print strings.map { |s| HASH.new s }.sum(&:value), "\n"

File: 15-1.py
#!/usr/bin/env python

def HASH(string):
    value = 0
    for c in string:
        value = ((value + ord(c)) * 17) % 256
    return value


input = open('15.input').readlines()[0].strip().split(",")
print(sum([HASH(s) for s in input]))
